httpclient/httpclient.go

```diff
@@ -183,28 +191,30 @@ func (c *HTTPConnectDialer) DialContext(ctx context.Context, network, address st
 		c.cacheH2Mu.Unlock()
 	}
 
-	rawConn, err := c.Dialer.DialContext(ctx, network, c.ProxyUrl.Host)
-	if err != nil {
-		return nil, err
-	}
-
+	var err error
+	var rawConn net.Conn
 	negotiatedProtocol := ""
 	switch c.ProxyUrl.Scheme {
 	case "http":
+		rawConn, err = c.Dialer.DialContext(ctx, network, c.ProxyUrl.Host)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Do you need a `break` here?

Reply:
Thankfully, Golang is not C, so I don't need a `break`.